lib/src/json_schema/custom_vocabularies.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:json_schema/src/json_schema/typedefs.dart';
+
+/// Register a custom vocabulary with the [JsonSchema] compiler.
+class CustomVocabulary {
+  CustomVocabulary(this._vocab, this._setters);
+
+  Uri _vocab;
+  Map<String, KeywordProcessor> _setters;
+
+  Uri get vocab => _vocab;
+
+  Map<String, KeywordProcessor> get setters => _setters;
+}
+
+/// A class to contain the set of functions for setting and validating keywords in a custom vocabulary.
+class KeywordProcessor {
+  KeywordProcessor(this._setter, this._validator);
+
+  SchemaPropertySetter _setter;
+  CustomValidationResult Function(Object, Object) _validator;
```

Comment:
Do you want to name these `Objects`?
```suggestion
  CustomValidationResult Function(Object attribute, Object validator) _validator;
```

Reply:
Could typedef alias this to: `KeywordValidator` and rename `SchemaPropertySetter` --> `KeywordSetter` if it's not public?

Reply:
I was going with the recommendation here: https://dart.dev/guides/language/effective-dart/design#prefer-inline-function-types-over-typedefs

I'm not sure I agree with the recommendation for this case though

Reply:
After looking at the output from `dartdoc`, it looks nicer to spell out the long function name instead of a typedef. (IMHO).

Reply:
Yeah, this is a case where I'd advise breaking that suggestion. I'm normally a fan of defining them inline until things get too confusing.